pkg/shipper/shipper.go

```diff
@@ -120,41 +115,20 @@ func New(
 // to remote if necessary, including compacted blocks which are already in filesystem.
 // It attaches the Thanos metadata section in each meta JSON file.
 func NewWithCompacted(
-	ctx context.Context,
 	logger log.Logger,
 	r prometheus.Registerer,
 	dir string,
 	bucket objstore.Bucket,
 	lbls func() labels.Labels,
 	source metadata.SourceType,
-	prometheusURL *url.URL,
-) (*Shipper, error) {
+) *Shipper {
 	if logger == nil {
 		logger = log.NewNopLogger()
 	}
 	if lbls == nil {
 		lbls = func() labels.Labels { return nil }
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
What's the reason for getting this out of shipper? It's is kind of a logical step before running shipper. Just curious. (:

Reply:
Do we still need this compaction check in shipper? Since it is already checked in sidecar.

Reply:
Definitely not, would be nice to have it one place, question is WHAT place. 

I think what you have now is OK.